browser/src/docevents.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+    This is for listening to / firing document events.
+    Will be useful for custom events but native events should also be bound via this bridge for maintainability.
+    Search "app.events.on" and "app.events.fire" for examples.
+
+    For firing custom events:
+        app.events.fire('myCustomEvent', {..details});
+*/
+
+type eventCallback = (...args: any[]) => void;
+
+class DocEvents {
+    static documentContainerId = 'document-container';
+    container: HTMLDivElement;
+
+    public initiate() {
+        this.container = document.getElementById(DocEvents.documentContainerId) as HTMLDivElement;
+
+        if (!this.container)
+            console.error('DocEvents initation failed.');
+
+        // Resize event for a div element is not fired. We'll use a resize-observer for catching the event.
+        new ResizeObserver(this.fire.bind(this, 'resize')).observe(this.container);
+    }
+
+    public fire(eventType: string, details: any) {
+        const newEvent = new CustomEvent(eventType, details);
+        this.container.dispatchEvent(newEvent);
+    }
+
+    public on(eventType: string, callback: eventCallback) {
```

Comment:
we need also `off` I think

Reply:
I couldn't decide on this. Not sure if we have a real need for off functionality here. Let's keep that out for now, if we see a scenario that needs, then we can always add it quickly.